website/versioned_docs/version-5.13.0/standardized-functions.md

```diff
@@ -1,5 +1,223 @@
+- **Streamlining the margin calculation process**: Simplify the steps involved in calculating initial margins, making it easier for market participants to follow
+
+These objectives will collectively enhance the effectiveness and reliability of margin calculations in the financial industry.
+
+# Methodology for CDM Implementation
+
+The methodology for implementing the standardized margin schedule in the CDM framework consists of three main steps:
+
+1. **Identification of required inputs for calculation**: The first step involves **identifying and gathering all necessary data points** that will be used to compute the initial margin. This ensures that all relevant factors are considered in the calculation process.
+2. **Extraction of trade information for initial margin calculation**: Dedicated functions are utilized to **extract relevant trade data**, ensuring that the calculations are based on accurate and up-to-date information.
+3. **Calculation of initial margin amounts**: Finally, the standardized **formulas and methodologies are applied to determine the required initial margin** accurately, taking into account the specific characteristics of each trade.
+
+This systematic approach promotes **clarity** and **accuracy** in margin calculations, ensuring compliance with regulatory standards while enhancing risk management practices.
+
+## Identification of inputs
+
+To compute the initial margin using the standardized schedule method, specific inputs need to be identified from each trade. The following inputs are crucial:
+
+1. **Asset Class**: The category of the derivative (e.g., credit, equity, commodity) that influences the margin rate.
+2. **Product Class**: A specific classification of the financial product, aligned with ISDA guidelines to ensure standardization.
+3. **Notional Amount**: The total value of the derivative contract, representing the size of the exposure.
+4. **Notional Currency**: The currency in which the notional amount is denominated, necessary for financial calculations.
+5. **Duration (in years)**: The remaining time until the derivative contract matures, which impacts the risk profile and margin requirement.
+
+Using the **concept of product class** helps ensure consistency in determining the notional amount and duration for each type of product. This standardization is vital for harmonizing calculations across different market participants and enhancing comparability.
+
+## Extraction of Information
+
+Function `BuildStandardizedSchedule ` is designed to **extract all relevant trade information** necessary for calculating the gross initial margin. It streamlines the process of gathering data from trades, ensuring accuracy and efficiency.
+
+``` Haskell
+func BuildStandardizedSchedule:
+    inputs: 
+        trade Trade (1..1)
+    output: standardizedSchedule StandardizedSchedule (1..1)
+    alias assetClass:
```

Comment:
Just a suggestion, but by including the entire function in the documentation you are obligating yourself to update the documentation whenever the function changes. Where functions have been described in similar documents we tend to just put the function name, inputs and output to prevent this maintenance overhead.